packages/components/src/breadcrumbs/test/index.tsx

```diff
@@ -109,4 +109,46 @@ describe( 'Breadcrumbs', () => {
 			screen.queryByRole( 'button', { name: 'More breadcrumb items' } )
 		).not.toBeInTheDocument();
 	} );
+
+	it( 'renders a custom item link', async () => {
+		const user = userEvent.setup();
+		const consoleLog = jest.fn();
+		render(
+			<Breadcrumbs
+				items={ FIVE_ITEMS }
+				renderItemLink={ ( { label, href, ...props } ) => {
+					const onClick = ( event: React.MouseEvent< HTMLAnchorElement > ) => {
+						props.onClick?.( event );
+						event.preventDefault();
+						consoleLog( `clicked ${ label }` );
+					};
+					return (
+						<a href={ href } { ...props } onClick={ onClick }>
+							{ label }
+						</a>
+					);
+				} }
+				variant="compact"
+				showCurrentItem
+			/>
+		);
+		const firstItemLabel = FIVE_ITEMS[ 0 ].label;
+		await user.click( screen.getByRole( 'link', { name: firstItemLabel } ) );
+		expect( consoleLog ).toHaveBeenCalledTimes( 1 );
+		expect( consoleLog ).toHaveBeenCalledWith( `clicked ${ firstItemLabel }` );
+
+		// The renderProp should not be applied to the current item.
+		// We have two items with the same label due to the internal implementation
+		// for calculating when to render as 'compact'.
+		await user.click( screen.getAllByText( FIVE_ITEMS[ FIVE_ITEMS.length - 1 ].label )[ 1 ] );
```

Comment:
I'm not sure if there is a better way to test this..

Reply:
I think it's alright - I found it clear enough.